core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -794,6 +794,19 @@ private[spark] class TaskSetManager(
             fetchFailed.bmAddress.host, fetchFailed.bmAddress.executorId))
         }
 
+        // Kill any other attempts for this FetchFailed task
+        for (attemptInfo <- taskAttempts(index) if attemptInfo.running) {
+          logInfo(s"Killing attempt ${attemptInfo.attemptNumber} for task ${attemptInfo.id} " +
+            s"in stage ${taskSet.id} (TID ${attemptInfo.taskId}) on ${attemptInfo.host} " +
+            s"as the attempt ${info.attemptNumber} failed because FetchFailed")
+          killedByOtherAttempt(index) = true
+          sched.backend.killTask(
```

Comment:
if this is async, we can't guarantee to not have task success events after marking staging as failed, right?

Reply:
This should not work. Maybe we shall just ignore the finished tasks submitted to a failed stage?

Reply:
@cloud-fan  Yes you're right, I should guarantee this in TaskSetManager.

Reply:
I don't think so. Useless tasks should fail soon(FetchFailure usually means mapper is down).

Reply:
Got it, I remove the code and UT in next commit.